### DISCUSSION TITLE
Where to put `fuzziness` or `prefix_length` in the request?

### DISCUSSION THREAD

**Discussion by AnLuoRidge** [zinclabs/zinc#601] [Q&A]
Hi there, 

Does anyone know where to put `fuzziness` or `prefix_length` in the request `POST /api/:target/_search`? 
I failed to find the usage example in API doc. Only found some ES samples in Issues, but they are not helpful.

Following is the request I've tried/guessed. Tested it with different `fuzziness` and `prefix_length`, but it won't work.

```json
{
    "search_type": "fuzzy",
    "query": {
        "term": "daughter",
        "prefix_length": 0,
        "fuzziness": 10
    },
    "sort_fields": ["weight"],
    "from": 0,
    "max_results": 20,
    "_source": [
       "en"
    ],
    "prefix_length": 0,
    "fuzziness": 10
}
```

The sentence I'm looking for is `the colonel's anger at his daughter's disobedience.`

I reckon I got 0 results because the distance between  `daughter's` and `daughter` is 2 and the default distance is 1. I didn't place `fuzziness` or `prefix_length` in the right place.

If I search by `daughters` or `daughter'`, I can get the correct result.

```json
{
    "took": 0,
    "timed_out": false,
    "_shards": {
        "total": 3,
        "successful": 3,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": {
            "value": 1
        },
        "max_score": 0.35508484912835664,
        "hits": [
            {
                "_index": "sample",
                "_type": "_doc",
                "_id": "1SIZZtGxb4A",
                "_score": 0.35508484912835664,
                "@timestamp": "2022-10-12T09:24:43.783059712Z",
                "_source": {
                    "en": "the colonel's anger at his daughter's disobedience."
                }
            }
        ]
    }
}
```

Thank you for your help.

**Comment by hengfeiyang** [comment]
You should use ElasticSearch compatible API.

https://docs.zincsearch.com/api-es-compatible/search/search/

**Reply by AnLuoRidge** [reply]
Thank you! I've checked this doc in fact but there are no samples for `fuzziness` or `prefix_length`. Should I read ES's doc?

**Reply by hengfeiyang** [reply]
Yes, we haven't written the detail documents, you can use ES doc for it.
